stock_inventory/models/stock_inventory.py

```diff
@@ -191,24 +191,55 @@ def _get_domain_category_quants(self, base_domain):
             ]
         )
 
+    def _get_products(self):
+        self.ensure_one()
+        if self.product_selection == "all":
+            return self.env["product.product"].search([])
+        elif self.product_selection == "category" and self.category_id:
+            return self.env["product.product"].search(
+                [("categ_id", "=", self.category_id.id)]
+            )
+        else:
+            return self.product_ids
+
+    def _get_overlapping_inventories(self):
+        for rec in self:
+            inventories = self.search([("state", "=", "in_progress")])
+            inventory_products = rec._get_products()
+
+            overlapping_inventories = inventories.filtered(
+                lambda x: (
+                    x.id != rec.id
```

Comment:
```suggestion
                    x.id != rec.id
```

you could remove this if when doing the search you already do like: [("state", "=", "in_progress"), ("id", "!=", rec.id)]

Reply:
I did this change, but if then I move the search out of the loop, so now this filtered is needed again

Reply:
removed the loop, and added this change again